perllib/FixMyStreet/Roles/CobrandEcho.pm

```diff
@@ -657,4 +658,67 @@ sub bulky_nice_item_list {
     ];
 }
 
+sub send_bulky_payment_echo_update_failed {
+    my ( $self, $params ) = @_;
+
+    # 3 hours to allow for Echo downtime
+    my $dtf = FixMyStreet::DB->schema->storage->datetime_parser;
+    my $cutoff_date
+        = $dtf->format_datetime( DateTime->now->subtract( hours => 3 ) );
+
+    my $rs = FixMyStreet::DB->resultset('Problem')->search(
+        {   category => 'Bulky collection',
+            cobrand  => $self->moniker,
+            created  => { '<'  => $cutoff_date },
+            extra    => { '\?' => 'payment_reference' },
+            -not => { extra => { '\?' => 'echo_update_failure_email_sent' } },
+        },
+    );
+
+    while ( my $report = $rs->next ) {
+        # Ignore if there is an update in Comment table with an external_id
+        next
+            if $report->comments->search( { external_id => { '!=', undef } },
+        )->count > 0;
```

Comment:
This is going to check every successful Bulky collection report each time it's run, even if it's looked at it before. We should set a flag here as well so that it doesn't have to do that, I think.